test/SILOptimizer/dead_store_elim.sil

```diff
@@ -231,21 +231,6 @@ bb0:
   return %4 : $()
 }
 
-// CHECK-LABEL: sil @store_after_store
-// CHECK: bb0
-// CHECK: {{ store}}
-// CHECK-NOT: {{ store}}
-// CHECK: return
-sil @store_after_store : $@convention(thin) (@owned B) -> () {
-bb0(%0 : $B):
-  %1 = alloc_box $<τ_0_0> { var τ_0_0 } <B>
-  %1a = project_box %1 : $<τ_0_0> { var τ_0_0 } <B>, 0
-  store %0 to %1a : $*B
```

Comment:
We lose this optimization when the source is a reference type. I think that's a rare case but, I would like to add this back as a special-case optimization in another patch.